js/index.js

```diff
@@ -52,22 +52,59 @@ closePopupBtn.addEventListener('click', () => {
   // body.style.overflowY = 'visible';
 });
 
-const email = document.getElementById('email');
 const form = document.querySelector('.contact-form');
+const formName = document.getElementById('name');
+const formEmail = document.getElementById('email');
+const formMessage = document.getElementById('message');
 const errorMessage = document.getElementById('error-message');
 
 form.addEventListener('submit', (e) => {
   const messages = [];
   const isUpperCase = (string) => /[A-Z]/.test(string);
 
-  if (isUpperCase(email.value) === true) {
+  if (isUpperCase(formEmail.value) === true) {
     messages.push('Email must be in lowercase!');
     errorMessage.innerText = messages.join(', ');
     e.preventDefault();
   }
-  if (email.value === '' || email.value === null) {
+  if (formEmail.value === '' || formEmail.value === null) {
     messages.push('Email field can not be empty!');
     errorMessage.innerText = messages.join(', ');
     e.preventDefault();
   }
 });
+
+// Preserve form data on local storage
+const formData = {
+  name: '',
+  email: '',
+  text: '',
+};
+
+function setData() {
+  const dataEntry = JSON.parse(localStorage.getItem('dataForm'));
+  formName.value = dataEntry.name;
+  formEmail.value = dataEntry.email;
+  formMessage.value = dataEntry.text;
+}
+
+function saveData() {
+  formData.name = formName.value;
+  formData.email = formEmail.value;
+  formData.text = formMessage.value;
+
+  const toStringData = JSON.stringify(formData);
+  localStorage.setItem('formData', toStringData);
+
+  setData();
+}
+
+if (!localStorage.getItem('formData')) {
+  saveData();
+} else {
+  setData();
+}
```

Comment:
- [ ]  Please implement the feedback your previous reviewer suggested, Your form should save the data in localstorage, and the last submitted data should persist in your form.  I am trying to submit the form, but the data are not being saved in localstorage.

Reply:
> [ ] Please implement the feedback your previous reviewer suggested, Your form should save the data in localstorage, and the last submitted data should persist in your form. I am trying to submit the form, but the data are not being saved in localstorage.

@ishimwezachee Been trying to and i am still not  getting it. I am just left with this task. can you please describe in  more details.